src/app/community/create/create.component.html

```diff
@@ -1,35 +1,49 @@
 <mat-card class="communityCreationForm">
-  <mat-card-title>Community Creation</mat-card-title>
+  <mat-card-title>Create Community</mat-card-title>
 
   <mat-card-content>
-    <mat-form-field id="communityNameField">
-      <input matInput placeholder="Name (shorthand form, used in URL)" [formControl]="nameFormControl" [(ngModel)]="nameText" maxlength="25" >
-      <mat-error *ngIf="nameFormControl.hasError('required')">
-        Name is <strong>required</strong>
-      </mat-error>
-    </mat-form-field>
+    <form [formGroup]="form">
+      <mat-form-field>
+        <input matInput type="text" placeholder="Name (shorthand form, used in URL)" formControlName="name" maxlength="25" pattern="[a-zA-Z0-9]*">
+        <mat-error *ngIf="form.get('name').hasError('required')">
+          Name is <strong>required</strong>
+        </mat-error>
+        <mat-error *ngIf="form.get('name').hasError('minlength')">
+          Name must be at least 3 characters
+        </mat-error>
+        <mat-error *ngIf="form.get('name').hasError('pattern')">
+          The name can only contain alphanumeric characters without spaces
+        </mat-error>
+        <mat-error *ngIf="form.get('name').hasError('nameExists')">
+          That name is already taken
+        </mat-error>
+        <mat-hint *ngIf="form.get('name').valid">meme.place/p/{{form.get('name').value}} is available!</mat-hint>
+        <mat-hint align="end">Required</mat-hint>
+      </mat-form-field>
 
-    <mat-form-field id="communityTitleField">
-      <input matInput placeholder="Title" [formControl]="titleFormControl" [(ngModel)]="titleText" maxlength="100">
-      <mat-error *ngIf="titleFormControl.hasError('required')">
-        Community Title is <strong>required</strong>
-      </mat-error>
-    </mat-form-field>
+      <mat-form-field>
+        <input matInput placeholder="Title" formControlName="title" maxlength="100">
+        <mat-error *ngIf="form.get('title').hasError('required')">
+          Community Title is <strong>required</strong>
+        </mat-error>
+        <mat-hint align="end">Required</mat-hint>
+      </mat-form-field>
 
-    <mat-form-field id="comDescriptionField">
-      <textarea matInput placeholder="Description" [(ngModel)] = "descriptionText" rows="10"></textarea>
-    </mat-form-field>
+      <mat-form-field>
+        <textarea matInput placeholder="Description" formControlName="description" rows="3"></textarea>
+      </mat-form-field>
 
-    <mat-form-field id="sidebarTextField">
-      <input matInput placeholder="Sidebar Message" [(ngModel)]="sidebarText" maxlength="300">
-    </mat-form-field>
+      <mat-form-field>
+        <textarea matInput placeholder="Sidebar Message" formControlName="sidebar" rows="3"></textarea>
```

Comment:
why encourage the user to write long sidebar information? Shouldn't it be short? our side bar isn't that big even on web.

Reply:
From what I've seen on reddit, the sidebar text is very important since it contains the rules, guidelines, further info, etc... and as a result, is usually very long.

Reply:
But what we're doing here is different from reddit. It's not just a form for people to post comments and discuss. Majority will be browsing and just posting pictures and people will be reacting. I honestly don't see someone setting up a large set of rules/guidelines. But it's not a big deal, I just think it'll look bad on the side bar; even with the expansion panel.